Authenticator/Source/OTPTokenListViewController.swift

```diff
@@ -214,17 +209,43 @@ extension OTPTokenListViewController /* UITableViewDelegate */ {
     }
 
     override func tableView(tableView: UITableView, didSelectRowAtIndexPath indexPath: NSIndexPath) {
-        let keychainItem = self.tokenManager.keychainItemAtIndex(indexPath.row)
-        if self.editing {
-            editKeychainItem(keychainItem)
-        } else {
-            if let password = keychainItem.token.currentPassword {
-                UIPasteboard.generalPasteboard().setValue(password, forPasteboardType: kUTTypeUTF8PlainText as String)
-                SVProgressHUD.showSuccessWithStatus("Copied")
+        if let cell = tableView.cellForRowAtIndexPath(indexPath) as? TokenRowCell {
+            let rowModel = cell.rowModel
+            if self.editing {
+                if let action = rowModel?.editAction {
+                    handleAction(action)
+                }
+            } else {
+                if let action = rowModel?.selectAction {
+                    handleAction(action)
+                }
             }
         }
     }
 
+    // MARK: TokenRowDelegate
+
+    func handleAction(action: TokenRowModel.Action) {
+        switch action {
+        case .UpdateKeychainItem(let keychainItem):
+            updateKeychainItem(keychainItem)
+        case .CopyPassword(let password):
+            copyPassword(password)
+        case .EditKeychainItem(let keychainItem):
+            editKeychainItem(keychainItem)
+        }
+    }
+
+    private func updateKeychainItem(keychainItem: Token.KeychainItem) {
+        let newToken = updatedToken(keychainItem.token)
+        saveToken(newToken, toKeychainItem: keychainItem)
+    }
+
+    private func copyPassword(password: String) {
+        UIPasteboard.generalPasteboard().setValue(password, forPasteboardType: kUTTypeUTF8PlainText as String)
+        SVProgressHUD.showSuccessWithStatus("Copied")
```

Comment:
Length Violation: Line should be 100 characters or less: currently 110 characters